src/reducers/updateThoughts.ts

```diff
@@ -105,15 +105,14 @@ const updateThoughts = (state: State, { thoughtIndexUpdates, contextIndexUpdates
 
     // Reset cursor on first load. The pullQueue can determine which contexts to load from the url, but cannot determine the full cursor (with ranks) until the thoughts have been loaded. To make it source agnostic, we decode the url here.
     // state.cursor should always be null here, but check to make sure we're not overriding it
-    !state.cursorInitialized && !state.cursor
+    !state.cursor && thoughts
```

Comment:
There is a problem with checking `state.cursor` instead of `cursorInitialized`. null is the value expected when the cursor is not on any thought, e.g. when you hit escape enough times. Thus `cursor` can be null throughout regular use of the app, not only when it has not yet been initialized. This may solve the issue, but it results in this block of code being called redundantly instead of once on load. We need to initialize the cursor only once.

Reply:
Note that `thoughts` is always truthy, so `&& thoughts` is a noop.